src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -499,25 +498,38 @@ internal virtual void Validate()
 
         internal ApplicationConfiguration BuildConfiguration()
         {
+            ResolveAuthority();
             Validate();
             return Config;
         }
 
         #region Authority
-        private void CreateAuthorityInfoFromEnums()
+        private void ResolveAuthority()
         {
             if (Config.AuthorityInfo != null)
             {
-                return;
+                if (!string.IsNullOrEmpty(Config.TenantId) &&
```

Comment:
Does this also cover application configurations like "PublicClientApplicationOptions"?

Reply:
Seems so. I added a test to it.